homeassistant/components/mqtt/client.py

```diff
@@ -1130,7 +1130,7 @@ def _async_mqtt_on_message(
             " retained" if msg.retain else "",
             topic,
             msg.qos,
-            msg.payload[0:8192],
+            msg.payload[0:8192] if msg.payload is not None else "<NULL>",
```

Comment:
```suggestion
            msg.payload[0:8192],
```
This should never be `None`